tests/ui/lint/non-local-defs/generics.rs

```diff
@@ -1,8 +1,6 @@
 //@ check-pass
 //@ edition:2021
 
-#![warn(non_local_definitions)]
-
 trait Global {}
 
 fn main() {
```

Comment:
The case that follows below here is interesting, in that it's an example of something that someone could reasonably want to do that also doesn't *feel* like a "sneaky inner impl", and where (other than allowing the lint) we wouldn't be leaving someone a good option other than to promote the trait to a wider scope.

```rust
trait Global {}
fn main() {
    trait Local {}
    impl<T: Local> Global for Vec<T> { }
    //~^ WARN non-local `impl` definition
}
```

(This is just an observation.  This clearly has non-local effects in terms of coherence checking.)